raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2442,10 +2503,10 @@ private void maybeTransition(
                 " and epoch " + epoch + " which is inconsistent with current leader " +
                 quorum.leaderId() + " and epoch " + quorum.epoch());
         } else if (epoch > quorum.epoch()) {
-            if (leaderId.isPresent()) {
+            if (leaderId.isPresent() && !leaderEndpoints.isEmpty()) {
```

Comment:
I added this because I thought it was an oversight not to check for empty endpoints given that we can initialize in UnattachedState when the leaderendpoints are not known. But I see now that `maybeTransition` is only called in places where the leaderendpoints are expected to be populated